### DISCUSSION TITLE
Does this only work for charging through the dedicated power connector?

### DISCUSSION THREAD

**Discussion by darelldd** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#651] [Q&A]
I used AlDente with great success on my 2013 MBP. Now I have a 2021  M1 14" MBP and I'm using USBC to charge it (via my Thunderbolt from the monitor). And AlDente seems to just be completely ignored. Charges to 100% no matter where I set it. I want to pay for the Pro version.... but not until I can see that it will work for my new machine. The only thing I can think of is that it will only work for the Magsafe connector input power, and not the input power from the USBC ports?

Note that I have unchecked the Optimize Battery Charging in system prefs. Is there something else I missed?

**Comment by Taavi47** [comment]
I have a MacBook Pro  M1 14" 2021. The Aldente program works well with all power connectors.

**Reply by darelldd** [reply]
Thank you for taking the time to let me know that this should be working even for the Thunderbolt power inputs. This is the first time I've heard that confirmed. I've now tried the magsafe power input too.... and on my machine AlDente just doesn't seem to be working. What else could I be missing? The only required change on my machine is ensuring that Optimize Battery Charging is off in preferences, right? 

 I'll try yet another reboot.

**Reply by darelldd** [reply]
Well, I went ahead and purchased the "pro" so I can use the tech support. I've asked the question there. In the meantime, after inputting my activation code.... I don't get the pro settings! Though they did send me the support email contact, so there's that.

**Comment by darelldd** [comment]
Solution: My problem seems to have been entirely because my software was set to intel-based machine. I have no idea why. I had no idea how to change the setting. So I "reset" to defaults. And that solved it. Charge is no limited no matter if power comes from the magsafe or the thunderbolts. Yay!
